### ISSUE TITLE
🧩 [Phase 5] ⭐ Cross-Attention Fusion — Core Research Contribution

### ISSUE THREAD

**Issue by shoaib-inamdar** [shoaib-inamdar/MultiModal-Cane-Disease-Detection#7]
**Parent:** #2 

## 🎯 Goal
Implement the Cross-Attention mechanism that lets visual tokens (Q) attend to
environmental tokens (K, V) — the core novel contribution of this research.

```
Attn(Q, K, V) = softmax(QKᵀ / √d) · V
```

## 📋 Files to Create
- [ ] `src/models/fusion/cross_attention.py` — Multi-head cross-attention module
- [ ] `src/models/multimodal_model.py` — Full model: encoder + fusion + classifier
- [ ] `src/models/heads/disease_classifier.py` — Linear(128 → 5) + Softmax
- [ ] `tests/unit/test_cross_attention.py`

## ✅ Done When
```python
output = model(batch["image"], batch["metadata"])
assert output.shape == torch.Size([B, 5])         # 5 disease classes
# loss.backward() completes without NaN
# attention weights sum ≈ 1.0
```
